tuw_nlp/graph/utils.py

```diff
@@ -49,7 +49,8 @@ def node_matcher(n1, n2):
         if n1['name'] is None or n2['name'] is None:
             return True
 
-        return True if re.match(fr"\b({n2['name']})\b", n1['name'], re.IGNORECASE) else False
+        return True if (re.match(fr"\b({n2['name']})\b", n1['name'], re.IGNORECASE) or n2['name'] == n1['name']) \
```

Comment:
@GKingA why do you need this second clause? doesn't it imply the first?

Reply:
I added it so the KnowledgeNode.__eq__ function will get called if we use the KnowledgeGraph (which is needed so the synsets/name of synsets can be compared).
The first use is the standard used for every other graph, so this way I don't disturb the normal usage.

Reply:
but you're comparing the name fields, aren't those strings?

Reply:
The *name* is just the name of the data part of the networkx node. In this case the *name* field is a KnowledgeNode object, in case of a 4lang or UD graph this would indeed be a string.

Reply:
I see. I'm concerned that there is code that assumes that this is a string, either here or in potato... @adaamko what do you think? @GKingA what is the structure in a KnowledgeNode exactly? Can't they be represented as strings?

Reply:
The KnowledgeNode is a subclass of string, but the __eq__ has been overridden, so codes assuming it to be a string work fine (at least in my experience).
However I did have to add an additional line to the code where the pickle file is unpickled when using these graphs:
sys.modules["knowledge_graph"] = knowledge_graph

KnowledgeNode has six attributes: lemma (str), text (str), pos (str), synset (Synset), antonym (Synset), conceptnet (Iterable of Concepts).

Reply:
Oh OK, I see, thanks for the info!